src/main/java/ForceDiscardPhase.java

```diff
@@ -30,11 +31,12 @@ public Phase execute() {
     }
 
     void discard(Player player, JsonArray jsonArray) {
-        JsonObject jsonObject = jsonArray.get(0).getAsJsonObject();
-
-        for (Resource resource : Constants.ALL_RESOURCES) {
-            if (jsonObject.has(resource.name().toLowerCase())) {
-                player.removeResources(resource, jsonObject.get(resource.name().toLowerCase()).getAsInt());
+        for (JsonElement obj : jsonArray) {
+            JsonObject object = obj.getAsJsonObject();
+            for (Resource resource : Constants.ALL_RESOURCES) {
+                if (object.get("type").getAsString().toLowerCase().equals(resource.name().toLowerCase())) {
```

Comment:
Dit checkt of de opgegeven naam bestaat toch? Is het niet beter hier een functie voor te maken?